__tests__/integration.js

```diff
@@ -0,0 +1,60 @@
+const Application = require("spectron").Application;
+const assert = require("assert");
+const path = require("path");
+const fs = require("fs");
+
+const appPath = "nteract/applications/desktop/dist/squashfs-root/nteract";
+const cleanNotebookPath = "notebooks/node-example.ipynb";
+const executedNotebookPath = "notebooks/node-example-executed.ipynb";
+
+const delay = time => new Promise(resolve => setTimeout(resolve, time));
+
+jest.setTimeout(15000);
+
+let nextTestNb = executedNotebookPath;
+describe("Testing notebook actions in python with dirty notebook", () => {
+  beforeEach(() => {
+    app = new Application({
+      path: appPath,
+      args: [nextTestNb]
+    });
+    return app.start();
+  });
+
+  afterEach(async () => {
+    if (app && app.isRunning()) {
+      await delay(2000);
+      return app.stop();
+    }
+  });
+
+  it("Open dirty notebook example, clears all cells", async () => {
+    await delay(12000);
+    await app.client.windowByIndex(0);
+    await app.browserWindow.send("menu:clear-all");
+    await app.browserWindow.send("menu:save-as", "notebooks/clear-all.ipynb");
+    // Setting the path for the starting notebook for the next test. Inelegant for sure, but it works ¯\_(ツ)_/¯
+    nextTestNb = cleanNotebookPath;
+  });
+
+  it("Open clean notebook example, runs all cells", async () => {
+    await delay(12000);
+    await app.client.windowByIndex(0);
+    await app.browserWindow.send("menu:run-all");
+    await app.browserWindow.send("menu:save-as", "notebooks/run-all.ipynb");
```

Comment:
Hmmmm.... We probably need to watch for when the notebook is "done" executing. We'd probably have to hook into the store to see when it's idle I assume (?)

Reply:
That's a good idea! -- I wasn't sure the best approach to get away from using `--forceExit` 😅

Reply:
I was thinking mostly about how `run-all` the command will finish before the cells are really done running, which means the save as will be missing the outputs.

Reply:
Ohhhh gotcha! I hadn't thought of that. Will try to incorporate in the future 👍